pkg/api/password.go

```diff
@@ -25,7 +25,8 @@ func SendResetPasswordEmail(c *m.ReqContext, form dtos.SendResetPasswordEmailFor
 
 	emailCmd := m.SendResetPasswordEmailCommand{User: userQuery.Result}
 	if err := bus.Dispatch(&emailCmd); err != nil {
-		return Error(500, "Failed to send email", err)
+		c.Logger.Error("Failed to send password reset email for ", "user", userQuery.LoginOrEmail)
+		return Error(200, "Email sent", err)
```

Comment:
This seems like very wrong return message

Reply:
I don't mind what message you decide is best for the moment, I just don't want to leak information based on username availability. The return code could equally be 500, I'm not sure the caller does anything with it.

The issue as to whether this is a 'real' issue or 'other people do it', doesn't help me, the pen testers and presumably malicious attackers do gain information which could trivially be stopped. 

I await your suggestions for a return code and a suitable message and I will amend the PR with them.

Reply:
You need to check for user not found error and handle that the same was as sent success, and all other errors should work as before (500 status code with error message)

Reply:
I have changed the return code to 500, the user not found check is already done in the preceding lines.

Reply:
I cannot see a check for that error, it's done like this:
https://github.com/grafana/grafana/blob/dafcfd70a709f0111ccb6b3bd74d4659c3f51a99/pkg/api/user.go#L38

Reply:
A change like that looks a lot different than the code already there. I don't know if I can commit to supporting such extensive changes. The issue is that the messages sent back to the client are different, I do not know the possible states that the user may or may not be in.

Reply:
Of course it needs to be different if you want to isolate the user not found case from any possible error.  

I can close this an you can open an issue for the underlying problem instead if you feel unsure how to fix it.

Reply:
I don't want to isolate any issues, that is the current problem, it does isolate and display different messages, I just wanted the same error message displayed so that user information doesn't leak.

Will an issue stand any more chance of fixing this issue in any reasonable timeframe?